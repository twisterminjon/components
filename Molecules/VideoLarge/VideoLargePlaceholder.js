import React from 'react'
import PropTypes from 'prop-types'
import useUser from '../../../hooks/useUser'

import Avatar from '../../Atoms/Avatar/Avatar'

import './VideoLarge.css'

VideoLargePlaceholder.propTypes = {
  /** User id of user to query */
  userId: PropTypes.string,
}

export default function VideoLargePlaceholder({ userId = '' }) {
  const { profileImage, displayName, loading } = useUser(userId)

  return <VideoLargePlaceholderView profileImage={profileImage} displayName={displayName} loading={loading} />
}

export function VideoLargePlaceholderView({ profileImage, displayName, loading }) {
  return (
    <div className="videolarge-placeholder-wrap">
      <Avatar loading={loading} displayName={displayName} size="massive" imgUrl={profileImage} />
    </div>
  )
}
