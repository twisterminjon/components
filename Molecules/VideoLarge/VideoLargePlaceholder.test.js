import React from 'react'
import { VideoLargePlaceholderView } from './VideoLargePlaceholder'

describe('VideoLargePlaceholder', () => {
  it('matches the base snapshot', () => {
    const wrapper = window.shallow(<VideoLargePlaceholderView />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot with loading', () => {
    const wrapper = window.shallow(<VideoLargePlaceholderView loading={true} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot with profile image', () => {
    const wrapper = window.shallow(<VideoLargePlaceholderView profileImage={'https://www.fillmurray.com/400/400'} />)
    expect(wrapper).toMatchSnapshot()
  })
})
