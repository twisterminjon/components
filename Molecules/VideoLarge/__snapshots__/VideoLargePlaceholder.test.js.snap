// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoLargePlaceholder matches the base snapshot 1`] = `
<div
  className="videolarge-placeholder-wrap"
>
  <Avatar
    className=""
    dimmed={false}
    displayName={null}
    imgUrl=""
    loading={false}
    size="massive"
    style={Object {}}
  />
</div>
`;

exports[`VideoLargePlaceholder matches the snapshot with loading 1`] = `
<div
  className="videolarge-placeholder-wrap"
>
  <Avatar
    className=""
    dimmed={false}
    displayName={null}
    imgUrl=""
    loading={true}
    size="massive"
    style={Object {}}
  />
</div>
`;

exports[`VideoLargePlaceholder matches the snapshot with profile image 1`] = `
<div
  className="videolarge-placeholder-wrap"
>
  <Avatar
    className=""
    dimmed={false}
    displayName={null}
    imgUrl="https://www.fillmurray.com/400/400"
    loading={false}
    size="massive"
    style={Object {}}
  />
</div>
`;
